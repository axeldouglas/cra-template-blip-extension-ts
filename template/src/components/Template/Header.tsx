import { useTranslation } from 'react-i18next';

import settings from 'config';

function Template() {
    const { t } = useTranslation();

    return (
        <header className="px-4 mx-auto sm:px-4 lg:mx-11">
            <div className="flex justify-between items-center pt-6 pb-4 border-b border-neutral-medium-wave md:justify-start md:space-x-3">
                <div className="flex justify-start items-center lg:flex-1 lg:w-0">
                    <h1 className="text-2xl font-semibold text-neutral-dark-city">
                        {t('title.homePage')}
                    </h1>
                    <div
                        className="pt-2 ml-2 text-neutral-medium-cloud cursor-help"
                        data-testid="page-header-tooltip"
                    >
                        <bds-tooltip
                            position="right-center"
                            tooltip-text="BLiP extension it's a technology-agnostic way to plug unofficial features to enhance the portal capabilities."
                        >
                            <bds-icon name="info" theme="solid">
                                Hover me
                            </bds-icon>
                        </bds-tooltip>
                    </div>
                </div>
                <div className="hidden justify-end items-center md:flex md:flex-1 lg:w-0">
                    <bds-button
                        variant="primary"
                        icon="link"
                        onClick={() =>
                            window.open(settings.repositoryUrl, '_blank')
                        }
                    >
                        {t('button.repository')}
                    </bds-button>
                </div>
            </div>
        </header>
    );
}

export default Template;
