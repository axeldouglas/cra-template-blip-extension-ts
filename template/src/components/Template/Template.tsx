import { useEffect } from 'react';
import { useLocation, Outlet } from 'react-router-dom';
import { page } from 'services/analytics-service';
import Header from './Header';

function Template() {
    const { pathname, state } = useLocation();

    // Analytics
    useEffect(() => {
        page({ pathname, state });
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [pathname]);

    return (
        <div className="relative">
            <Header />
            <Outlet />
        </div>
    );
}

export default Template;
