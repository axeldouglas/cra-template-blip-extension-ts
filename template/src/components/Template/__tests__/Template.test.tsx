import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';

import settings from 'config';

import Template from '..';

describe('<Template />', () => {
    const mockedOpen = jest.fn();
    const originalOpen = window.open;

    beforeAll(() => {
        window.open = mockedOpen;
    });

    afterAll(() => {
        // Restore original
        window.open = originalOpen;
    });

    it('should navigate to repository', async () => {
        const user = userEvent.setup();

        render(<Template />, { wrapper: MemoryRouter });

        const repositoryButton = screen.getByText(/button.repository/i);
        expect(repositoryButton).toBeInTheDocument();

        await user.click(repositoryButton);
        expect(mockedOpen).toHaveBeenCalledTimes(1);
        expect(mockedOpen).toHaveBeenCalledWith(
            settings.repositoryUrl,
            '_blank'
        );
    });

    it('should render the Template', () => {
        const { container } = render(<Template />, { wrapper: MemoryRouter });

        expect(container.firstChild).toMatchSnapshot();
    });
});
