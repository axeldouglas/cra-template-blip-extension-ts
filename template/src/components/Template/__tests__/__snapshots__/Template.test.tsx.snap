// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Template /> should render the Template 1`] = `
<div
  class="relative"
>
  <header
    class="px-4 mx-auto sm:px-4 lg:mx-11"
  >
    <div
      class="flex justify-between items-center pt-6 pb-4 border-b border-neutral-medium-wave md:justify-start md:space-x-3"
    >
      <div
        class="flex justify-start items-center lg:flex-1 lg:w-0"
      >
        <h1
          class="text-2xl font-semibold text-neutral-dark-city"
        >
          title.homePage
        </h1>
        <div
          class="pt-2 ml-2 text-neutral-medium-cloud cursor-help"
          data-testid="page-header-tooltip"
        >
          <bds-tooltip
            position="right-center"
            tooltip-text="BLiP extension it's a technology-agnostic way to plug unofficial features to enhance the portal capabilities."
          >
            <bds-icon
              name="info"
              theme="solid"
            >
              Hover me
            </bds-icon>
          </bds-tooltip>
        </div>
      </div>
      <div
        class="hidden justify-end items-center md:flex md:flex-1 lg:w-0"
      >
        <bds-button
          icon="link"
          variant="primary"
        >
          button.repository
        </bds-button>
      </div>
    </div>
  </header>
</div>
`;
